### PR TITLE
find decoding candidates for logs

### PR DESCRIPTION
**Pull request by ayrat555**
resolves https://github.com/poanetwork/blockscout/issues/2521

## Changelog
- find decoding candidates for logs

### CODE REVIEW

File: apps/block_scout_web/lib/block_scout_web/templates/address_logs/_logs.html.eex [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
+    <%= case decoded_result do %>
+      <%= {:error, :contract_not_verified, _cadidates} -> %>
+        <div class="alert alert-info">
+        <%= gettext "To see accurate decoded input data, the contract must be verified." %>
+        <%= case @transaction do %>
+          <% %{to_address: %{hash: hash}} -> %>
+            <%= gettext "Verify the contract " %><a href="<%= address_verify_contract_path(@conn, :new, hash)%>"><%= gettext "here" %></a>
+          <% _ -> %>
+            <%= nil %>
+            <% end %>
```
Comment (pasqu4le) [inline apps/block_scout_web/lib/block_scout_web/templates/address_logs/_logs.html.eex]: This line seems to have too much indentation

File: apps/block_scout_web/lib/block_scout_web/templates/transaction_log/_logs.html.eex [hunk 2]
```diff
@@ -1,4 +1,20 @@
...
+  <%= case decoded_result do %>
+    <%= {:error, :contract_not_verified, _cadidates} -> %>
+      <div class="alert alert-info">
+      <%= gettext "To see accurate decoded input data, the contract must be verified." %>
+      <%= case @transaction do %>
+        <% %{to_address: %{hash: hash}} -> %>
+          <%= gettext "Verify the contract " %><a href="<%= address_verify_contract_path(@conn, :new, hash)%>"><%= gettext "here" %></a>
+        <% _ -> %>
+          <%= nil %>
+          <% end %>
```
Comment (pasqu4le) [inline apps/block_scout_web/lib/block_scout_web/templates/transaction_log/_logs.html.eex]: Same indentation issue here

File: apps/explorer/lib/explorer/chain/log.ex [hunk 3]
```diff
@@ -114,7 +115,44 @@ defmodule Explorer.Chain.Log do
          do: {:ok, identifier, text, mapping}
   end
 
-  def decode(_log, _transaction), do: {:error, :contract_not_verified}
+  def decode(log, transaction) do
+    case log.first_topic do
+      "0x" <> hex_part ->
+        {number, ""} = Integer.parse(hex_part, 16)
```
Comment (pasqu4le) [inline apps/explorer/lib/explorer/chain/log.ex]: Could this fail to produce a `number` (or leaving `""`)?
If so I think that is a decoding error and should be returned like the other `case`: `{:error, :could_not_decode}`, right?

File: apps/block_scout_web/lib/block_scout_web/templates/address_logs/_logs.html.eex (hunk 1 above)
Comment (ayrat555) [reply apps/block_scout_web/lib/block_scout_web/templates/address_logs/_logs.html.eex]: thanks

File: apps/block_scout_web/lib/block_scout_web/templates/transaction_log/_logs.html.eex (hunk 2 above)
Comment (ayrat555) [reply apps/block_scout_web/lib/block_scout_web/templates/transaction_log/_logs.html.eex]: thanks

File: apps/explorer/lib/explorer/chain/log.ex (hunk 3 above)
Comment (ayrat555) [reply apps/explorer/lib/explorer/chain/log.ex]: make sense

**Comment by ayrat555** [comment]
@vbaranov I reverted these changes

**Pull request merged**